Headers/AppKit/NSPasteboard.h

```diff
@@ -296,14 +316,54 @@ APPKIT_EXPORT NSString *NSPasteboardCommunicationException;
 @end
 
 #endif
-//
-// Return File-related Pasteboard Types
-//
 APPKIT_EXPORT NSString *NSCreateFileContentsPboardType(NSString *fileType);
 APPKIT_EXPORT NSString *NSCreateFilenamePboardType(NSString *fileType);
 APPKIT_EXPORT NSString *NSGetFileType(NSString *pboardType);
 APPKIT_EXPORT NSArray *NSGetFileTypes(NSArray *pboardTypes);
 
+#if OS_API_VERSION(MAC_OS_X_VERSION_10_6, GS_API_LATEST)
+typedef NSUInteger NSPasteboardWritingOptions;
+enum {
+  NSPasteboardWritingPromised = 1 << 9
+};
+
+@protocol NSPasteboardWriting <NSObject>
+- (NSArray *)writableTypesForPasteboard:(NSPasteboard *)pasteboard;
```

Comment:
Shouldn't these methods go into the optional part? Otherwise we end up with a missing @end. The same is true for Reading.